lib/rouge/lexers/php.rb

```diff
@@ -153,11 +164,12 @@ def self.detect?(text)
           end
         end
 
-        rule %r/(\d+\.\d*|\d*\.\d+)(e[+-]?\d+)?/i, Num::Float
-        rule %r/\d+e[+-]?\d+/i, Num::Float
-        rule %r/0[0-7]+/, Num::Oct
-        rule %r/0x[a-f0-9]+/i, Num::Hex
-        rule %r/\d+/, Num::Integer
+        rule %r/(?:(?:#{LNUM}|#{DNUM})[eE][+-]?#{LNUM})/o, Num::Float
+        rule %r/#{DNUM}/o, Num::Float
+        rule %r/0[0-7]+(?:_[0-7]+)*/, Num::Oct
+        rule %r/0[bB][01]+(?:_[01]+)*/, Num::Bin
```

Comment:
If you add the `i` flag at the end, you don't need to separately list `b` and `B`.

Reply:
If you prefer the `i` modifier, I'll change that ;)